src/Atc.Azure.IoT/Models/IIoTHubModuleServiceModels.cs

```diff
@@ -37,6 +39,67 @@ public record UploadSupportBundleRequest(
     [property: JsonPropertyName("edgeRuntimeOnly")] bool EdgeRuntimeOnly = false)
     : SchemaVersion10;
 
+/// <summary>
+/// Retrieve module logs and upload to Azure Blob Storage.
+/// </summary>
+/// <param name="SasUrl">Shared Access Signature URL with write access to Azure Blob Storage container.</param>
+/// <param name="Items">An array with id and filter tuples.</param>
+/// <param name="Encoding">Either gzip or none. Default is none.</param>
+/// <param name="ContentType">Either json or text. Default is text.</param>
+public record UploadModuleLogsRequest(
+    [property: JsonPropertyName("sasUrl")] Uri SasUrl,
+    [property: JsonPropertyName("items")] IEnumerable<LogRequestItem> Items,
+    [property: JsonPropertyName("encoding")] LogContentEncoding Encoding = LogContentEncoding.None,
```

Comment:
LogContentEncoding contains gzip and not encoding values - are we talking compression or encoding?

Reply:
Encoding https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/Content-Encoding